### ISSUE TITLE
Getting a warning when updating viewState using onViewStateChange 

### ISSUE THREAD

**Issue by malzaeem-sw** [visgl/deck.gl#4358]
#### Description

Getting a warning after updating to react 16.13.0 

> Warning: Cannot update a component from inside the function body of a different component.

You can find more details about this warning [here](https://reactjs.org/blog/2020/02/26/react-v16.13.0.html) 

Also, the warning doesn't show when I don't use onViewStateChange 

#### Repro Steps

Try to update viewState from onViewStateChange like the following
```
<DeckGL
  layers={layers}
  viewState={viewState}
  controller={true}
  onViewStateChange={({viewState})=>setViewState(viewState)}
/>
```

#### Environment (please complete the following information):
 - Framework Version: [e.g. deck.gl 8.0.17]
 - Browser Version: [e.g. Chrome 80.0.3987.132]
 - OS: Windows 10

**Comment by Pessimistress** [comment]
Please include your definition of `setViewState`. We cannot reproduce the issue from the code you shared.

**Comment by malzaeem** [comment]
Sorry, it's just a hook set function like so:

 `const [viewState, setViewState] = useState(initialViewState);`

**Comment by yinshanyang** [comment]
I too am facing the same issue, though the difference on my side is that the warning is only issued when I zoom using a double-click action. Zooming by mousewheel works fine.

 - Framework Version: deck.gl 8.1.0
- Browser Version: Firefox 74.0 & Chrome 80.0.3987.149
- OS: MacOS 10.14.6

**Comment by Pessimistress** [comment]
This will be fixed in 8.5.0. You can try it right now with `8.5.0-alpha.1`.
